src/bp/core/messaging/messaging-router.ts

```diff
@@ -0,0 +1,38 @@
+import * as sdk from 'botpress/sdk'
+import { UnauthorizedError } from 'common/http'
+import { HTTPServer } from 'core/app/server'
+import { CustomRouter } from 'core/routers/customRouter'
+import { Router } from 'express'
+import { MessagingLegacy } from './legacy'
+import { MessagingService } from './messaging-service'
+
+export class MessagingRouter extends CustomRouter {
+  private legacy: MessagingLegacy
+
+  constructor(private logger: sdk.Logger, private messaging: MessagingService, private http: HTTPServer) {
+    super('Messaging', logger, Router({ mergeParams: true }))
+    this.legacy = new MessagingLegacy(logger, http)
+  }
+
+  public setupRoutes(): void {
+    this.router.post('/receive', async (req, res, next) => {
```

Comment:
Please add the async middleware so the server doesn't crash if there is an exception in messaging